tsl/test/expected/data_node.out

```diff
@@ -1813,17 +1830,103 @@ ORDER BY 1, 2;
+\set ON_ERROR_STOP 0
+ALTER TABLE hyper3 ADD COLUMN temp2 int;
+ERROR:  some data nodes are not available for DDL queries
+ALTER TABLE hyper_1dim ADD COLUMN temp2 int;
+ERROR:  some data nodes are not available for DDL queries
+\set ON_ERROR_STOP 1
+-- Mark all DNs unavailable. Metadata should still retain last DN but all
+-- activity should fail
+SELECT * FROM alter_data_node('data_node_2', available=>false);
+WARNING:  could not switch data node on 2 chunks
+  node_name  |   host    | port  |    database    | available 
+-------------+-----------+-------+----------------+-----------
+ data_node_2 | localhost | 55432 | db_data_node_2 | f
+(1 row)
+
+SELECT * FROM alter_data_node('data_node_3', available=>false);
+WARNING:  could not switch data node on 11 chunks
+  node_name  |   host    | port  |    database    | available 
+-------------+-----------+-------+----------------+-----------
+ data_node_3 | localhost | 55432 | db_data_node_3 | f
+(1 row)
+
+\set ON_ERROR_STOP 0
+INSERT INTO hyper3 VALUES ('2022-01-10 00:00:00', 1, 1);
+ERROR:  insufficient number of data nodes
+INSERT INTO hyper_1dim VALUES ('2022-01-10 00:00:00', 1, 1);
+ERROR:  insufficient number of data nodes
+UPDATE hyper3 SET temp = 10 WHERE time = '2022-01-03 00:00:00';
+ERROR:  insufficient number of data nodes
+UPDATE hyper_1dim SET temp = 10 WHERE time = '2022-01-03 00:00:00';
+ERROR:  insufficient number of data nodes
+DELETE FROM hyper3 WHERE time = '2022-01-03 00:00:00';
+ERROR:  insufficient number of data nodes
+DELETE FROM hyper_1dim WHERE time = '2022-01-03 00:00:00';
+ERROR:  insufficient number of data nodes
+SELECT count(*) FROM hyper3;
+ERROR:  Node "data_node_3" is not available for queries
+SELECT count(*) FROM hyper_1dim;
+ERROR:  Node "data_node_3" is not available for queries
+ALTER TABLE hyper3 ADD COLUMN temp2 int;
+ERROR:  some data nodes are not available for DDL queries
+ALTER TABLE hyper_1dim ADD COLUMN temp2 int;
+ERROR:  some data nodes are not available for DDL queries
+\set ON_ERROR_STOP 1
 -- re-enable the data node and the chunks should "switch back" to
 -- using the data node. However, only the chunks for which the node is
 -- "primary" should switch to using the data node for queries
 ALTER DATABASE data_node_1_unavailable RENAME TO :DN_DBNAME_1;
 WARNING:  you need to manually restart any running background workers after this command
 SELECT * FROM alter_data_node('data_node_1', available=>true);
+WARNING:  insufficient number of data nodes
+WARNING:  insufficient number of data nodes
   node_name  |   host    | port  |    database    | available 
 -------------+-----------+-------+----------------+-----------
  data_node_1 | localhost | 55432 | db_data_node_1 | t
 (1 row)
 
+SELECT * FROM alter_data_node('data_node_2', available=>true);
+WARNING:  insufficient number of data nodes
+WARNING:  insufficient number of data nodes
+WARNING:  insufficient number of data nodes
+WARNING:  insufficient number of data nodes
```

Comment:
It's a bit odd to see 4 of the same WARNING. How many can there be in worst case?  Is this one per stale chunk? That could be quite jarring if many chunks are stale. Can we fix?

Reply:
My concern here remains. Maybe it is related to something different?

Reply:
@erimatnor this is not from this PR. It's due to the existing behavior of `alter_data_node('data_node_2', available=>true);`

It's coming from `ts_hypertable_assign_chunk_data_nodes` when we iterate over chunks to remove `data_node_2` as the primary DN for those many chunks.

Reply:
Got it.